pkgs/tools/typesetting/tex/texlive/build-texlive-package.nix

```diff
@@ -0,0 +1,187 @@
+{ lib
+, fetchurl
+, runCommand
+
+  # script interpreters
+, bash
+, jdk
+, perl
+, python3
+, ruby
+, snobol4
+, tk
+
+  # TeX Live prerequisites
+, texliveBinaries
+}:
+
+{ pname
+, revision
+, version ? toString revision
+, extraRevision ? ""
+, extraVersion ? ""
+, sha512
+, mirrors
+, fixedHashes ? { }
+, postUnpack ? ""
+, stripPrefix ? 1
+, license ? [ ]
+, hasHyphens ? false
+, hasInfo ? false
+, hasManpages ? false
+, hasRunfiles ? false
+, hasTlpkg ? false
+, ...
+}@args:
+
+/* buildTeXLivePackage converts an attribute set extracted from tlpdb.nix (with
+   the deps attribute already processed) to a fake multi-output derivation with
+   possible outputs [ "tex" "texdoc" "texsource" "tlpkg" "out" "man" "info" ]
+*/
+
+let
+  name = "${pname}-${version}${extraVersion}";
+
+  # common metadata
+  meta = {
+    license = map (x: lib.licenses.${x}) license;
+    # TeX Live packages should not be installed directly into the user profile
+    outputsToInstall = [ ];
+  };
+
+  # emulate the outputSpecified attribute
+  metaSpecified = meta // { outputSpecified = true; };
```

Comment:
![image](https://github.com/NixOS/nixpkgs/assets/1962985/2f319cb6-fb8d-4159-87db-f01530046e61)

This doesn't pass CI! Interesting quirk of `meta`. So I have to smuggle `meta.outputSpecified` *outside* of the derivation (or in `passthru`, if that works).